external/fv3fit/fv3fit/pytorch/graph/graph_builder.py

```diff
@@ -1,46 +1,57 @@
-import torch
+from typing import Tuple
 import numpy as np
-import geopy.distance
-from vcm.catalog import catalog
 import dataclasses
+from fv3fit.keras._models.shared.halos import append_halos
+import xarray as xr
 
 
 @dataclasses.dataclass
 class GraphConfig:
     """
     Attributes:
-        neighbor: number of nearest neighbor grids
-        coarsen: 1 if the full model resolution is used, othewise data will be coarsen
-        resolution: Model resolution to load the corresponding lat and lon.
+        nx_tile: number of horizontal grid points on each tile of the cubed sphere
     """
 
-    neighbor: int = 10
-    coarsen: int = 8
-    resolution: str = "grid/c48"
+    # TODO: this should not be configurable, it should be determined
+    # by the shape of the input data
+    nx_tile: int = 48
 
 
-def build_graph(config: GraphConfig) -> tuple:
-    nodes = []
-    edges = []
-
-    grid = catalog[config.resolution].read()
-    lat = grid.lat.load()
-    lon = grid.lon.load()
-
-    lat = lat[:, :: config.coarsen, :: config.coarsen].values.flatten()
-    lon = lon[:, :: config.coarsen, :: config.coarsen].values.flatten()
+def build_graph(nx_tile: int) -> Tuple[np.ndarray, np.ndarray]:
```

Comment:
Unfortunately the number of edges can't change - it can't be fewer, because then a direction must be missing. If it is more, the increased edges must be symmetric, which would mean adding 4 corners to 9 edges, but for corner points there are only 8 neighbor points, not 9. If DGL supports having different numbers of edges per node, we can add an `include_corners: bool` argument for this later.